### PR TITLE
Add cliplugins to framework meta package and Make clusterclass package consistent with other packages

### PR DESCRIPTION
**Pull request by anujc25**
### What this PR does / why we need it
- Add `cliplugins` to `framework` composite package 
- Make `clusterclass` package consistent with other packages

### Which issue(s) this PR fixes

Related #1996

### Describe testing done for PR

```
make management-package-bundles
make package-repo-bundle PACKAGE_REPOSITORY=management
make push-management-package-bundles
make push-management-package-repo-bundle
```

```
# install package repository
tanzu package repository update tanzu-management --create --namespace tkg-system --url gcr.io/eminent-nation-87317/tkg/test5/management:v0.21.0-dev

# install tkg composite package
tanzu package install tkg-pkg --package-name tkg.tanzu.vmware.com --version 0.21.0-dev --values-file ~/Documents/code/tanzu-framework/packages/management/tkg/testdata/sample-values.yaml --namespace tkg-system
```

### Release note

```release-note
package-based-lcm: Add cliplugins to framework meta package
```

### PR Checklist

- [x] Squash the commits into one or a small number of logical commits
      
- [x] Use good commit [messages](https://github.com/vmware-tanzu/tanzu-framework/blob/main/CONTRIBUTING.md)
- [x] Ensure PR contains terms all contributors can understand and links all contributors can access

### CODE REVIEW

**Review by yharish991 — APPROVED** [review]
lgtm

File: packages/management/tkg/bundle/config/packageinstalls/tkg-clusterclass-pi.yaml [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+kind: Secret
+metadata:
+  name: tkg-clusterclass-values
+  namespace: #@ data.values.clusterclassPackage.namespaceForPackageInstallation
+  annotations:
+    kapp.k14s.io/change-rule.0: "upsert after upserting tkg-clusterclass-packageinstall/serviceaccount"
+    kapp.k14s.io/change-rule.1: "delete before deleting tkg-clusterclass-packageinstall/serviceaccount"
+type: Opaque
+stringData:
+  values.yaml: #@ yaml.encode(clusterclassPackageValues())
```
Comment (prkalle) [inline packages/management/tkg/bundle/config/packageinstalls/tkg-clusterclass-pi.yaml:42]: suggestion: you can replace `clusterclassPackageValues()` method with `data.values.clusterclassPackage`

**Review by prkalle — APPROVED** [review]
LGTM. Thanks!

**Review by vuil — APPROVED** [review]
lgtm, thanks!

File: packages/management/tkg/bundle/config/packageinstalls/tkg-clusterclass-pi.yaml (hunk 1 above)
Comment (anujc25) [reply packages/management/tkg/bundle/config/packageinstalls/tkg-clusterclass-pi.yaml:42]: There is minor differences between what `clusterclassPackageValues()` returns and what we have as part of `data.values.clusterclassPackage` and hence I have added this function.

**Comment by yharish991** [comment]
@anujc25 since we are calling `composite` package as `meta` package now, can you also update the release note to reflect that?

**Pull request merged**